packer/builder/azure/smapi/step_create_image.go

```diff
@@ -88,7 +88,8 @@ func (s *StepCreateImage) Cleanup(state multistep.StateBag) {
 				return vmdisk.NewClient(client).DeleteDisk(diskName, true)
 			}, retry.ConstantBackoffRule("busy", func(err management.AzureError) bool {
 				return strings.Contains(err.Message, "is currently performing an operation on deployment") ||
-					strings.Contains(err.Message, "is currently in use by virtual machine")
+					strings.Contains(err.Message, "is currently in use by virtual machine") ||
+					strings.Contains(err.Message, "connection reset by peer")
```

Comment:
The error that would match this is not a management.AzureError, so this condition will never be true...
Sorry to disappoint...;-)